src/dataExplorer/components/SaveAsNotebookForm.tsx

```diff
@@ -0,0 +1,171 @@
+// Libraries
+import React, {FC, ChangeEvent, useState} from 'react'
+import {useSelector} from 'react-redux'
+import {useHistory} from 'react-router-dom'
+
+// Selectors
+import {getActiveTimeMachine, getSaveableView} from 'src/timeMachine/selectors'
+import {getOrg} from 'src/organizations/selectors'
+import {postNotebook} from 'src/client/notebooksRoutes'
+
+// Components
+import {Form, Input, Button, Grid} from '@influxdata/clockface'
+import {AUTOREFRESH_DEFAULT} from 'src/shared/constants'
+
+// Types
+import {
+  Columns,
+  InputType,
+  ButtonType,
+  ComponentColor,
+  ComponentStatus,
+} from '@influxdata/clockface'
+
+// Utils
+import {event, normalizeEventName} from 'src/cloud/utils/reporting'
+import {chartTypeName} from 'src/visualization/utils/chartTypeName'
+import {
+  PROJECT_NAME,
+  DEFAULT_PROJECT_NAME,
+  PROJECT_NAME_PLURAL,
+} from 'src/flows'
+import {hydrate, serialize} from 'src/flows/context/flow.list'
+
+interface Props {
+  dismiss: () => void
+}
+
+const SaveAsNotebookForm: FC<Props> = ({dismiss}) => {
+  const [notebookName, setNotebookName] = useState('')
+  const orgID = useSelector(getOrg).id
+  const stuff = useSelector(getActiveTimeMachine)
+  const {draftQueries, autoRefresh, timeRange} = stuff
+  const {properties} = useSelector(getSaveableView)
+  const history = useHistory()
+
+  const handleChangeNotebookName = (event: ChangeEvent<HTMLInputElement>) => {
+    setNotebookName(event.target.value)
+  }
+
+  const handleSubmit = async () => {
+    event(`Data Explorer Save as ${PROJECT_NAME} Submitted`)
+
+    try {
+      event(`data_explorer.save.as_${PROJECT_NAME.toLowerCase()}.success`, {
+        which: normalizeEventName(chartTypeName(properties?.type ?? 'xy')),
+      })
+      const pipes: any = []
+
+      const visualization = {
+        title: 'Visualize the Result',
+        visible: true,
+        type: 'visualization',
+        properties,
+      }
+      for (let i = 0; i < draftQueries.length; i++) {
+        const draftQuery = draftQueries[i]
+
+        let pipe: any = {
+          family: 'inputs',
+          visible: !draftQuery.hidden,
+        }
+        if (draftQuery.editMode === 'builder') {
+          pipe.name = 'Build a Query'
+          pipe.type = 'queryBuilder'
+          pipe = {
+            ...pipe,
+            ...draftQuery.builderConfig,
+            buckets: [
+              {
+                // id: TODO(ariel): get the id
+                name: draftQuery.builderConfig.buckets[0],
+                orgID,
+                type: 'user',
+              },
+            ],
+          }
+        } else {
+          pipe.title = 'Query to Run'
+          pipe.queries = properties.queries
+          pipe.activeQuery = i
```

Comment:
idk how well this is supported in notebooks. it's _kind of_ in there, but we never exposed the interface to switch between queries so there prolly haven't been a lot of eyes on it for a bit. might be safer to always run `activeQuery = 0; queries = properties.queries[i]`

Reply:
sweet. Good to know